### DISCUSSION TITLE
Neverending Electron download

### DISCUSSION THREAD

**Discussion by RuiCuco** [Kode/Kha#1414] [General]
Hi. It's been a while. Hope you're all okay and free of virus.

Now, whenever I start VSCode, Kha starts to download Electron, and it never stops. I can-t press F5 as usual to debug, because it says it can-t find Electron. I tried using Krom instead, but VSCode complains that targeted API must be 2 and it is 1.

Am I overlooking something here_

**Comment by RobDangerous** [comment]
See https://github.com/Kode/vscode-kha/issues/36 - from what I can tell so far a recent vscode update broke things, got another report like that yesterday. You can use the workaround Moritz mentioned until it's fixed. For Krom to work make sure Kha and Krom are up to date (aka don't use the versions in the extension, download both from GitHub and set the paths in the extension options).

**Comment by RobDangerous** [comment]
Should be working again now.

**Comment by RuiCuco** [comment]
Yes, it's working. Thanks. Only now it gives the error
<img width="439" alt="Capture" src="https://user-images.githubusercontent.com/16723900/162828424-75fb5d0b-cccd-461e-ac89-51e9a37aca3e.PNG">
every time I hit F5. The only way to not give this error is to erase the files and folders in the build directory first. When I do that, the debug session proceeds as normal when I hit F5.

**Comment by RobDangerous** [comment]
Doesn't happen to me. Please provide more information - like, much much more information.

**Comment by RuiCuco** [comment]
Do you have the line

"debug.onTaskErrors": "debugAnyway"

in your VS Code settings.json? If so, please remove it. Once that is done, please remove the files from your current project build folder and hit F5. It should run okay. The next time you hit F5 (with the build folder already populated), the error I mentioned appears. This is what the terminal output shows:

> Executing task: node D:\Developer\Kha\make.js debug-html5 --debug <

Using Kha (944ff74d) from D:\Developer\Kha
Creating Kha project.
Exporting asset 1 of 25 (Title.png).
........
Exporting asset 25 of 25 (IconCatalog.png).
Compiling shader 1 of 2 (main.frag.glsl).
Compiling shader 2 of 2 (main.vert.glsl).
Compiling shader 1 of 8 (painter-colored.frag.glsl).
Compiling shader 2 of 8 (painter-colored.vert.glsl).
Compiling shader 3 of 8 (painter-image.frag.glsl).
Compiling shader 4 of 8 (painter-image.vert.glsl).
Compiling shader 5 of 8 (painter-text.frag.glsl).
Compiling shader 6 of 8 (painter-text.vert.glsl).
Compiling shader 7 of 8 (painter-video.frag.glsl).
Compiling shader 8 of 8 (painter-video.vert.glsl).
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command node D:\Developer\Kha\make.js debug-html5 --debug" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.

**Comment by RobDangerous** [comment]
As I said, doesn't happen to me and I never had "debugAnyway" in my settings. Do you also get it to return exit code 1 when you run "node D:\Developer\Kha\make.js debug-html5 --debug" on a regular command line? It should print "Done" at the end if it succeeds.

**Comment by RuiCuco** [comment]
In a command prompt, it does not print 'Done' at the end. The error only occurs when the build folder has files and folders related to html5 build. When there are no such items in the build folder, it runs okay either in VS Code (it debugs) or in a command prompt (it prints 'Done').

**Comment by RobDangerous** [comment]
First time something like that is reported. Can you open Kha/Tools/khamake in vscode, adjust args and/or in .vscode/launch.json to point to your project and hit F5? That might show you what's happening. And does that also happen with the Kha samples like for example https://github.com/Kha-Samples/Shader?

**Comment by RuiCuco** [comment]
Right, sorry for taking so long to reply (this is just a hobby for me, and I don't have much time in my hands).

It also happens with Kha samples, like Shader and Mesh Loader. If I erase the folder Build/_ProjectName_-debug-html5-intellij before hitting F5, things work okay, otherwise the error I mentioned occurs. (By the way, I also tried erasing the folder Build/debug-html5-resources, but Kha goes into an infinite loop, and nothing ever happens again.)

**Comment by RobDangerous** [comment]
No worries, it's also just a hobby for me. So, what about my first question?

**Comment by RuiCuco** [comment]
I tried searching for 'intellij' in various files inside khamake, as that is where the error happens, but I couldn't find it. Perhaps I didn't come across the right file. Could you please point me towards all the ???.js in khamake where that folder is created and populated?

**Comment by RobDangerous** [comment]
The error happens in intellij? Are we still talking about the same thing?

**Comment by RuiCuco** [comment]
Sorry, I guess I didn't explain myself very well. I thought you were keeping up with me. What I meant to say was: if the folder

Build/_ProjectName_-debug-html5-intellij

is present when I hit F5 in VSCode to debug, the error I mentioned occurs. If that folder is missing, no error occurs. So, I went to the khamake folder and tried to search the various *.js for 'intellij' to try to find out where that folder is created and populated. However, I had no luck finding that word. So, I'll repeat my question, could you please point me towards all the *.js in khamake where said folder is created and populated? That way, I may figure out what's happening.

**Comment by RobDangerous** [comment]
Ah, that's right here: https://github.com/Kode/khamake/blob/main/src/HaxeProject.ts#L28
Just set a breakpoint and step through, that should show you what's going on.

**Comment by RuiCuco** [comment]
Well, this is the error that Kha now prints (*** is the project name):

Error: EISDIR: illegal operation on a directory, readlink 'build\***-debug-html5-intellij\***.iml'
    at Object.readlinkSync (node:fs:1575:3)
    at checkDest (D:\Developer\Kha\Tools\khamake\node_modules\fs-extra\lib\copy-sync\copy-sync.js:184:23)
    at checkPaths (D:\Developer\Kha\Tools\khamake\node_modules\fs-extra\lib\copy-sync\copy-sync.js:208:24)
    at Object.copySync (D:\Developer\Kha\Tools\khamake\node_modules\fs-extra\lib\copy-sync\copy-sync.js:26:24)
    at IntelliJ (D:\Developer\Kha\Tools\khamake\out\HaxeProject.js:86:8)
    at writeHaxeProject (D:\Developer\Kha\Tools\khamake\out\HaxeProject.js:410:9)
    at exportProjectFiles (D:\Developer\Kha\Tools\khamake\out\main.js:148:44)
    at exportKhaProject (D:\Developer\Kha\Tools\khamake\out\main.js:545:22)
    at async exportProject (D:\Developer\Kha\Tools\khamake\out\main.js:553:16)
    at async Object.run (D:\Developer\Kha\Tools\khamake\out\main.js:726:16) {
  errno: -4068,
  syscall: 'readlink',
  code: 'EISDIR',
  path: 'build\\***-debug-html5-intellij\\***.iml'

Does it help?

**Comment by RobDangerous** [comment]
It fails to copy a file, looks like something is messed up on your hard disk. As I said before, just set a breakpoint, then you can see what's going on exactly - doesn't look to me like it's actually a problem with Kha.
